django/template/defaulttags.py

```diff
@@ -36,15 +36,22 @@ def __init__(self, setting, nodelist):
         self.setting, self.nodelist = setting, nodelist
 
     def render(self, context):
-        old_setting = context.autoescape
-        context.autoescape = self.setting
-        output = self.nodelist.render(context)
-        context.autoescape = old_setting
+        output = ''.join(self.stream(context))
         if self.setting:
             return mark_safe(output)
         else:
             return output
 
+    def stream(self, context):
+        old_setting = context.autoescape
+        context.autoescape = self.setting
+        for chunk in self.nodelist.stream(context):
+            if self.setting:
+                yield mark_safe(chunk)
```

Comment:
Since there's many occurrence of nested property access in the patch (`self.filter_expr.resolve`, ...) I don't think the micro-optimisation here is worth it without further benchmarking.